// nuxt-bits@0.1.6/dist/module.mjs
import { defineNuxtModule, createResolver, addVitePlugin, installModule, addComponentsDir, addImportsDir } from '@nuxt/kit';

const componentsFolders = ["Animations", "Backgrounds", "Components", "TextAnimations"];
const module = defineNuxtModule({
  meta: {
    name: "nuxt-bits",
    configKey: "bits",
    compatibility: {
      nuxt: ">=3.6.0"
    }
  },
  defaults: {
    prefix: "Bits"
  },
  async setup(options, nuxt) {
    const { resolve } = createResolver(import.meta.url);
    nuxt.options.css.push(resolve("./runtime/assets/styles.css"));
    if (nuxt.options.builder === "@nuxt/vite-builder") {
      const plugin = await import('@tailwindcss/vite').then((r) => r.default);
      addVitePlugin(plugin);
    } else {
      nuxt.options.postcss.plugins["@tailwindcss/postcss"] = {};
    }
    await installModule("@vueuse/nuxt");
    componentsFolders.forEach((folder) => {
      addComponentsDir({
        path: resolve(`./runtime/components/${folder}`),
        prefix: options.prefix,
        pathPrefix: false
      });
    });
    addImportsDir(resolve("./runtime/composables"));
  }
});

export { module as default };
